src/Simplic.Package.Service/FileService.cs

```diff
@@ -9,9 +10,56 @@ namespace Simplic.Package.Service
 {
     public class FileService : IFileService
     {
-        public async Task<byte[]> ReadAllBytesAsync(string path)
+        /// <summary>
+        /// Reads all bytes from a file at a given path
+        /// </summary>
+        /// <param name="path">The path to the file</param>
+        /// <returns>The read bytes as an array</returns>
+        public async Task<byte[]> ReadAllBytesAsync(string path) //TODO: Actually make async
         {
             return File.ReadAllBytes(path);
         }
+
+        /// <summary>
+        /// Reads all text from a file at a given path
+        /// </summary>
+        /// <param name="path">The path to the file</param>
+        /// <returns>The read text as a string</returns>
+        public async Task<string> ReadAllTextAsync(string path)
+        {
+            var byteArray = await ReadAllBytesAsync(path);
+
+            return Encoding.Default.GetString(byteArray);
+        }
+
+        /// <summary>
+        /// Reads all bytes from a stream
+        /// </summary>
+        /// <param name="stream">The stream to read from</param>
+        /// <returns>The read bytes as an array</returns>
+        public async Task<byte[]> ReadAllBytesAsync(Stream stream)
+        {
+            using (stream)
+            {
+                using (MemoryStream memoryStream = new MemoryStream())
+                {
+                    await stream.CopyToAsync(memoryStream);
+
+                    return memoryStream.ToArray();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Reads all text from a stream
+        /// </summary>
+        /// <param name="stream">The stream to read from</param>
+        /// <returns>The read text as a string</returns
+        public async Task<string> ReadAllTextAsync(Stream stream)
+        {
+            var byteArray = await ReadAllBytesAsync(stream);
+
+            return Encoding.Default.GetString(byteArray);
+        }
```

Comment:
.Net framework has also the possibility to use async file access:

https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/using-async-for-file-access